server/src/main/java/org/elasticsearch/transport/SniffConnectionStrategy.java

```diff
@@ -347,10 +347,16 @@ private void collectRemoteNodes(Iterator<Supplier<DiscoveryNode>> seedNodesSuppl
                 onFailure.accept(e);
             });
         } else {
-            listener.onFailure(new NoSeedNodeLeftException(strategyType(), clusterAlias));
+            listener.onFailure(getNoSeedNodeLeftException(Set.of()));
         }
     }
 
+    private NoSeedNodeLeftException getNoSeedNodeLeftException(Collection<Exception> suppressedExceptions) {
```

Comment:
Since we're only ever passing an empty set here, we could just remove the `suppressedExceptions` arg altogether.

Reply:
Since I have reduced scope of this PR. This method is no longer necessary. I inlined it which also removes the need of `suppressedExceptions`. Thanks for spotting it.